pkg/scheduler/objects/queue.go

```diff
@@ -1292,3 +1303,42 @@ func (sq *Queue) String() string {
 	return fmt.Sprintf("{QueuePath: %s, State: %s, StateTime: %x, MaxResource: %s}",
 		sq.QueuePath, sq.stateMachine.Current(), sq.stateTime, sq.maxResource)
 }
+
+func (sq *Queue) incRunningApps() {
```

Comment:
Should this increment and decrement below be using a lock on update?
Do need to make sure we do not lock the whole hierarchy, see incPendingResource for example.